File: frontend/lib/src/data/model/category/category_model.dart
import 'package:orizon/src/domain/entities/category/category_entity.dart';
import 'package:orizon/src/domain/entities/transaction/transaction_entity.dart';

class CategoryModel extends CategoryEntity {
  const CategoryModel({
    required super.id,
    required super.name,
    required super.type,
    super.icon,
    super.color,
    required super.createdAt,
  });

  factory CategoryModel.fromJson(Map<String, dynamic> json) {
    return CategoryModel(
      id: json['id'],
      name: json['name'],
      type: json['type'] == 'EXPENSE'
          ? TransactionType.expense
          : TransactionType.income,
      icon: json['icon'],
      color: json['color'],
      createdAt: DateTime.parse(json['createdAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'type': type == TransactionType.expense ? 'EXPENSE' : 'INCOME',
      'icon': icon,
      'color': color,
    };
  }
}

File: frontend/web/src/app/core/models/category.model.ts
import { TransactionType } from './transaction.model';

export interface CategoryResponse {
  id: string;
  name: string;
  type: TransactionType;
  icon: string | null;
  color: string | null;
  createdAt: string;
}

export interface CreateCategoryRequest {
  name: string;
  type: TransactionType;
  icon?: string;
  color?: string;
}

export interface UpdateCategoryRequest {
  name: string;
  type: TransactionType;
  icon?: string;
  color?: string;
}
